File: app/src/main/java/dizel/budget_control/budget/domain/BudgetApi.java
package dizel.budget_control.budget.domain;

import androidx.annotation.Nullable;

import com.google.firebase.database.PropertyName;

//TODO переписать на котлин

public class BudgetApi {
    @PropertyName("title")
    public String title = null;

    @PropertyName("currency")
    public String currency = null;

    private String sum = null;

    @PropertyName("budgetSum")
    public void setSum(Object sum) {
        if (sum instanceof Double) {
            this.sum = ((Double)sum).toString();
        } else if ( sum instanceof Long) {
            this.sum = ((Long)sum).toString();
        } else {
            this.sum = null;
        }
    }

    @Nullable
    @PropertyName("budgetSum")
    public String getSum() {
        return sum;
    }

}

File: app/src/main/java/dizel/budget_control/budget/domain/BudgetApi.kt
package dizel.budget_control.budget.domain

import com.google.firebase.database.PropertyName

data class BudgetApi(
    @get:PropertyName("budgetSum")
    @set:PropertyName("budgetSum")
    var sum: Long? = null,

    @get:PropertyName("currency")
    @set:PropertyName("currency")
    var currency: String? = null,

    @get:PropertyName("title")
    @set:PropertyName("title")
    var title: String? = null
)

data class CategoryApi(
    @get:PropertyName("categoryId")
    @set:PropertyName("categoryId")
    var id: String? = null,

    @get:PropertyName("categoryName")
    @set:PropertyName("categoryName")
    var name: String? = null,

    @get:PropertyName("categoryMoney")
    @set:PropertyName("categoryMoney")
    var money: Long? = null,

    @get:PropertyName("categoryCurrency")
    @set:PropertyName("categoryCurrency")
    var currency: String? = null,

    @get:PropertyName("categoryColor")
    @set:PropertyName("categoryColor")
    var color: String? = null

)